draft-ietf-tls-tls13.md

```diff
@@ -1566,7 +1566,9 @@ cipher_suites
   order of client preference. If the list contains cipher suites
   the server does not recognize, support, or wish to use, the server
   MUST ignore those cipher suites, and process the remaining ones as
-  usual. Values are defined in {{cipher-suites}}.
+  usual. Values are defined in {{cipher-suites}}. If the client is
+  attempting a PSK key establishment, it MUST only advertise the single
+  cipher suite associated with the PSK.
```

Comment:
I don't think I'm quite saying that. What the spec says is that you can't resume without the cipher suite having the same PRF:

"Any ticket MUST only be resumed with a cipher suite that has the same KDF hash as that used to establish the original connection, and if the client provides the same SNI value as described in Section 3 of [RFC6066]."

It's up to the server to implement this requirement however it chooses. As @davidben says, the way that NSS and BoringSSL implement this is to negotiate the cipher suite first (using whatever their conventional mechanism is, which may be server's preference, client's preference, or something else) and then condition resumption on whether the PSK has the appropriate KDF. Note that this cipher suite might be the client's last choice if, for instance, it's the server's first choice.